File: benchmarks/prime_sieve/Sieve.java
public class Sieve {
    public static void main(String[] args) {
        int N = 1000000;
        boolean[] sieve = new boolean[N];
        for (int i = 0; i < N; i++) sieve[i] = true;
        sieve[0] = false;
        sieve[1] = false;

        for (int i = 2; i * i < N; i++) {
            if (sieve[i]) {
                for (int j = i * i; j < N; j += i) {
                    sieve[j] = false;
                }
            }
        }

        int count = 0;
        for (int i = 0; i < N; i++) {
            if (sieve[i]) count++;
        }
        System.out.println(count);
    }
}

File: benchmarks/prime_sieve/sieve.py
N = 1000000
sieve = [True] * N
sieve[0] = False
sieve[1] = False

i = 2
while i * i < N:
    if sieve[i]:
        j = i * i
        while j < N:
            sieve[j] = False
            j += i
    i += 1

count = sum(1 for x in sieve if x)
print(count)
